pkg/metrics/metrics.go

```diff
@@ -351,20 +405,66 @@ func (c *Counter) IsAlertFiring(alert string, checkedNS, ignoredNS []string) (bo
 	return false, nil
 }
 
-func getPromHost(c client.Client) (*string, error) {
+// GetServiceEndpoint accepts a client,namespace,svcName and portName and attempts to retrive
+// the services endpoint in the form of resolveable.service:portnumber.
+func GetServiceEndpoint(c client.Client, namespace, svcName, portName string) (string, error) {
+	svc := &corev1.Service{}
+	err := c.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: svcName}, svc)
+	if err != nil {
+		return "", err
+	}
+
+	host := fmt.Sprintf(svcName + "." + namespace + clusterSVCSuffix)
+	var port string
+	for _, p := range svc.Spec.Ports {
+		if p.Name == portName {
+			port = strconv.FormatInt(int64(p.Port), 10)
+		}
+	}
+	endpoint := fmt.Sprint(host + ":" + port)
```

Comment:
Add some validation / error returning here if port remains an empty string having not been set.

Reply:
added validation. why is `Sprint()` redundant?

Reply:
Thanks for fixing, looks good.
re the `Sprint()` I meant it could be
`endpoint := host + ":" + port`
but that's a trivial matter.